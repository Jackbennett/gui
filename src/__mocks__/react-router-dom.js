import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
