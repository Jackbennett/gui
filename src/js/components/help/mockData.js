export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
