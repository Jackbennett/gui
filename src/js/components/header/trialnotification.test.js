import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { undefineds } from '../../../../tests/mockData';
import TrialNotification from './trialnotification';

describe('DeviceNotifications Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <TrialNotification />
      </MemoryRouter>
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });

  it('renders correctly with an expiration date', async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <TrialNotification expiration="2019-02-01T12:16:22.667Z" />
      </MemoryRouter>
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
