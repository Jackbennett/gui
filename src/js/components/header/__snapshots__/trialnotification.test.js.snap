// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceNotifications Component renders correctly 1`] = `
<div
  class="flexbox centered"
  id="trialVersion"
>
  <div
    class="muted margin-right-small"
    currentitem="false"
    data-event="click focus"
    data-for="trial-version"
    data-offset="{'bottom': 15, 'right': 60}"
    data-tip="true"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      style="margin-right: 2px; height: 16px; vertical-align: bottom;"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
    Trial plan
  </div>
  <div
    class="__react_component_tooltip place-bottom type-dark"
    data-id="tooltip"
    id="trial-version"
  >
    <h3>
      Trial plan
    </h3>
    <p>
      You're using the trial version of Mender – it's free for up to 10 devices for 12 months.
    </p>
    <p>
      <a
        href="/settings/upgrade"
      >
        Upgrade to a plan
      </a>
       to add more devices and continue using Mender after the trial expires.
    </p>
    <p>
      Or compare the plans at
       
      <a
        href="https://mender.io/plans/pricing"
        rel="noopener noreferrer"
        target="_blank"
      >
        mender.io/plans/pricing
      </a>
      .
    </p>
  </div>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    href="/settings/upgrade"
    id="trial-upgrade-now"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
          />
        </svg>
      </span>
      Upgrade now
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
</div>
`;

exports[`DeviceNotifications Component renders correctly with an expiration date 1`] = `
<div
  class="flexbox centered"
  id="trialVersion"
>
  <div
    class="muted margin-right-small"
    currentitem="false"
    data-event="click focus"
    data-for="trial-version"
    data-offset="{'bottom': 15, 'right': 60}"
    data-tip="true"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      style="margin-right: 2px; height: 16px; vertical-align: bottom;"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
    Trial plan
  </div>
  <div
    class="__react_component_tooltip place-bottom type-dark"
    data-id="tooltip"
    id="trial-version"
  >
    <h3>
      Trial plan
    </h3>
    <p>
      You're using the trial version of Mender – it's free for up to 10 devices for 12 months.
    </p>
    <p>
      <a
        href="/settings/upgrade"
      >
        Upgrade to a plan
      </a>
       to add more devices and continue using Mender after the trial expires.
    </p>
    <p>
      Or compare the plans at
       
      <a
        href="https://mender.io/plans/pricing"
        rel="noopener noreferrer"
        target="_blank"
      >
        mender.io/plans/pricing
      </a>
      .
    </p>
  </div>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    href="/settings/upgrade"
    id="trial-upgrade-now"
    role="button"
    tabindex="0"
  >
    <span
      class="MuiButton-label"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
          />
        </svg>
      </span>
      Upgrade now
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </a>
  <div
    class="muted"
  >
    You have 
    30
     
    days
     remaining on the trial plan
  </div>
</div>
`;
